ray-operator/controllers/ray/batchscheduler/yunikorn/yunikorn_scheduler.go

```diff
@@ -0,0 +1,70 @@
+package yunikorn
+
+import (
+	"context"
+	"github.com/go-logr/logr"
+	rayv1 "github.com/ray-project/kuberay/ray-operator/apis/ray/v1"
+	schedulerinterface "github.com/ray-project/kuberay/ray-operator/controllers/ray/batchscheduler/interface"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	SchedulerName                    string = "yunikorn"
+	PodApplicationIDLabelName        string = "applicationId"
+	PodQueueLabelName                string = "queue"
+	RayClusterApplicationIDLabelName string = "yunikorn.apache.org/application-id"
+	RayClusterQueueLabelName         string = "yunikorn.apache.org/queue-name"
+)
+
+type YuniKornScheduler struct {
+	log logr.Logger
+}
+
+type YuniKornSchedulerFactory struct{}
+
+func GetPluginName() string {
+	return SchedulerName
+}
+
+func (y *YuniKornScheduler) Name() string {
+	return SchedulerName
+}
+
+func (y *YuniKornScheduler) DoBatchSchedulingOnSubmission(_ context.Context, _ *rayv1.RayCluster) error {
+	// yunikorn doesn't require any resources to be created upfront
+	// this is a no-opt for this implementation
+	return nil
+}
+
+func (y *YuniKornScheduler) populatePodLabels(app *rayv1.RayCluster, pod *v1.Pod, sourceKey string, targetKey string) {
+	// check annotations
+	if value, exist := app.Labels[sourceKey]; exist {
+		y.log.Info("Updating pod label based on RayCluster annotations",
```

Comment:
Maybe persist `ctx` in `YuniKornScheduler`, and then use `ctrl.LoggerFrom(ctx)` to keep consistency.

Reply:
This code was similar to https://github.com/ray-project/kuberay/blob/ca39dc94f93592efc0ee7093b4fcded2a01cb7e5/ray-operator/controllers/ray/batchscheduler/volcano/volcano_scheduler.go#L176. But let me know if you insist on this.

Reply:
What does the log look like? Will this log message be in JSON format?

Reply:
It looks like below:

```
{"level":"info","ts":"2024-07-23T22:39:15.278Z","logger":"yunikorn","msg":"Updating pod label based on RayCluster annotations","sourceKey":"yunikorn.apache.org/application-id","targetKey":"applicationId","value":"my-ray-cluster-0001"}
```

Reply:
Nice. In this case, we don't need to update it in this PR.